tests/console/validate_dud_addon_repos.pm

```diff
@@ -0,0 +1,38 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate that DUD addon repos activated during the installation are
+# properly added and enabled. Also, verifies that 'zypper ref' works and all the
+# repositories can be refreshed.
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+
+use testapi;
+use repo_tools 'parse_repo_data';
+use scheduler 'get_test_suite_data';
+use Test::Assert ':all';
+
+sub run {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+    foreach my $expected_dud_repo (@{$test_data->{dud_repos}}) {
+        my $actual_dud_repo = parse_repo_data($expected_dud_repo->{URI});
+        assert_equals($expected_dud_repo->{Enabled}, $actual_dud_repo->{Enabled},
+            "Fail! It is expected that the 'Enabled' field is set to $expected_dud_repo->{Enabled}, but it is $actual_dud_repo->{Enabled}");
+        assert_equals($expected_dud_repo->{Autorefresh}, $actual_dud_repo->{Autorefresh},
+            "Fail! It is expected that the 'Autorefresh' field is set to $expected_dud_repo->{Autorefresh}, but it is $actual_dud_repo->{Autorefresh}");
+    }
```

Comment:
Despite i like the approach (clean and neat) i was wondering if `validate_repo_enablement` can be used here? Looking in the subroutine i think you could remove the extra test_data. But it might need some 'tweaking'. not sure. Do you think it would be better to do so?

Reply:
That was my initial plan. But `validate_repo_enablement` asserts equality of actual and expected URI by providing name or alias. But in the current test URI is used as repo identifier, because its alias and name are generated with the random postfix each test.

Also, I'm finding the approach with parsing the values from the output and storing them in a Hash reference a little bit more flexible while reusing the function. It just provides all the info about the specified repo, and then the test can decide of what to do with this data. Actually, I can refactor the existing tests and use the function instead of the `validate_repo_enablement`.

@rwx788 As you are the author of `validate_repo_enablement`, wouldn't you be against if I will replace it with `parse_repo_data` and assert_equals combination in `validate_addon_repos.pm` and `validate_mirror_repos.pm` in the scope of this PR?

Reply:
FYI i am also working in some refactoring in repos_tools. So to avoid conflicts we can either pair or leave the refactoring for later.

Reply:
Ah, ok. Then probably it makes sense to create a follow-up ticket in case we want to refactor the existing tests and do that in another PR. Just to not slow down the current work. 

Anyway, both functions work correctly and we can keep them untouched for any amount of time until we'll decide to refactor the existing code in case if we need to refactor it at all. Let's wait for @rwx788 opinion too.

Reply:
exactly.

Reply:
> That was my initial plan. But `validate_repo_enablement` asserts equality of actual and expected URI by providing name or alias. But in the current test URI is used as repo identifier, because its alias and name are generated with the random postfix each test.
> 
> Also, I'm finding the approach with parsing the values from the output and storing them in a Hash reference a little bit more flexible while reusing the function. It just provides all the info about the specified repo, and then the test can decide of what to do with this data. Actually, I can refactor the existing tests and use the function instead of the `validate_repo_enablement`.
> 
> @rwx788 As you are the author of `validate_repo_enablement`, wouldn't you be against if I will replace it with `parse_repo_data` and assert_equals combination in `validate_addon_repos.pm` and `validate_mirror_repos.pm` in the scope of this PR?

I've also have refactored it a bit, when worked on some repo validation. Feel free to improve it further. I guess you can find jobs I've used for validation to test your changes.